src/app/berkeley_account_tables/sql.ml

```diff
@@ -0,0 +1,479 @@
+
+            ON parent.id = b.parent_id
+
+            WHERE b.global_slot_since_genesis >= $1
+       |sql}
+
+  let run (module Conn : Caqti_async.CONNECTION) ~start_slot ~internal_cmd_id =
+    Conn.collect_list query (start_slot, internal_cmd_id)
+end
+
+module Public_key = struct
+  let query =
+    Caqti_request.find_opt Caqti_type.int Caqti_type.string
+      {sql| SELECT value FROM public_keys
+            WHERE id = ?
+      |sql}
+
+  let run (module Conn : Caqti_async.CONNECTION) pk_id =
+    Conn.find_opt query pk_id
+end
+
+module Snarked_ledger_hashes = struct
+  let query =
+    Caqti_request.find Caqti_type.int Caqti_type.string
+      {sql| SELECT value FROM snarked_ledger_hashes
+            WHERE id = ?
+      |sql}
+
+  let run (module Conn : Caqti_async.CONNECTION) id = Conn.find query id
+end
+
+module Epoch_data = struct
+  type epoch_data = { epoch_ledger_hash : string; epoch_data_seed : string }
+
+  let epoch_data_typ =
+    let encode t = Ok (t.epoch_ledger_hash, t.epoch_data_seed) in
+    let decode (epoch_ledger_hash, epoch_data_seed) =
+      Ok { epoch_ledger_hash; epoch_data_seed }
+    in
+    let rep = Caqti_type.(tup2 string string) in
+    Caqti_type.custom ~encode ~decode rep
+
+  let query_epoch_data =
+    Caqti_request.find Caqti_type.int epoch_data_typ
+      {sql| SELECT slh.value, ed.seed FROM snarked_ledger_hashes AS slh
+
+       INNER JOIN
+
+       epoch_data AS ed
+
+       ON slh.id = ed.ledger_hash_id
+
+       WHERE ed.id = ?
+
+      |sql}
+
+  let get_epoch_data (module Conn : Caqti_async.CONNECTION) epoch_ledger_id =
+    Conn.find query_epoch_data epoch_ledger_id
+
+  let query_staking_epoch_data_id =
+    Caqti_request.find Caqti_type.string Caqti_type.int
+      {sql| SELECT staking_epoch_data_id FROM blocks
+
+            WHERE state_hash = ?
+
+      |sql}
+
+  let get_staking_epoch_data_id (module Conn : Caqti_async.CONNECTION)
+      state_hash =
+    Conn.find query_staking_epoch_data_id state_hash
+
+  let query_next_epoch_data_id =
+    Caqti_request.find Caqti_type.string Caqti_type.int
+      {sql| SELECT next_epoch_data_id FROM blocks
+
+            WHERE state_hash = ?
+      |sql}
+
+  let get_next_epoch_data_id (module Conn : Caqti_async.CONNECTION) state_hash =
+    Conn.find query_next_epoch_data_id state_hash
+end
+
+module Parent_block = struct
+  (* fork block is parent of block with the given state hash *)
+  let query_parent_state_hash =
+    Caqti_request.find Caqti_type.string Caqti_type.string
+      {sql| SELECT parent.state_hash FROM blocks AS parent
+
+            INNER JOIN
+
+            (SELECT parent_id FROM blocks WHERE state_hash = ?) AS epoch_ledgers_block
+
+            ON epoch_ledgers_block.parent_id = parent.id
+      |sql}
+
+  let get_parent_state_hash (module Conn : Caqti_async.CONNECTION)
+      epoch_ledgers_state_hash =
+    Conn.find query_parent_state_hash epoch_ledgers_state_hash
+end
+
+module Balances = struct
+  let query_insert_nonce =
```

Comment:
How is this getting used?

Reply:
That's a leftover from the `compatible` replayer. I'll delete it.